app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RecipeFoodsController < ApplicationController
+  def new
+    @foods = Food.all
+    @current_recipe_id = params[:id]
+    @recipe_food = RecipeFood.new
+  end
+
+  def create
+    @recipe_food = RecipeFood.new(recipe_food_params)
+    if @recipe_food.save
+      redirect_to recipes_show_path(@recipe_food.recipe_id)
+    else
+      render 'new', status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @recipe_food = RecipeFood.find(params[:id])
+    @recipe = Recipe.find(@recipe_food.recipe_id)
+    @food = Food.find(@recipe_food.food_id)
+    @recipe_food.update(recipe_food_params)
+    redirect_to recipes_show_path(@recipe_food.recipe_id)
+  end
```

Comment:
### [OPTIONAL]

Overall, the code follows a standard approach commonly used in Rails applications. However, it's important to keep in mind best practices and consider potential improvements based on the specific requirements of your application. Some suggestions include:

Consider using Rails' built-in before_action or before_filter to reduce redundancy and improve code organization.
Look into using nested routes or associations to simplify the process of finding related records.
Validate the presence of required associations (recipe_id and food_id) to ensure data integrity.